### PR TITLE
mgr/cephadm: refresh service state in the background

### PR DESCRIPTION
**Pull request by liewegas**
- cache ssh connections
- refresh services in the background

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -646,13 +673,20 @@ def serve(self):
         # type: () -> None
         self.log.info("serve starting")
         while self.run:
+            self._check_hosts()
+
+            # refresh services
+            self.log.debug('refreshing services')
+            completion = self._get_services(maybe_refresh=True)
+            self._orchestrator_wait([completion])
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: #32893

**Comment by sebastian-philipp** [comment]
Fixes: https://tracker.ceph.com/issues/43679

File: src/pybind/mgr/cephadm/module.py (hunk 1 above)
Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: Yeah... I tried changing this based on your suggested change to the upgrade PR a few weeks back but I couldn't make it work, and you were out on PTO or something, so I went with this for now.  Can you try fixing it in a separate pR?

**Comment by liewegas** [comment]
retest this please

**Comment by sebastian-philipp** [comment]
I've pushed three commits to https://github.com/sebastian-philipp/ceph/commits/liewegas-wip-cephadm-services . They should fix the unit tests and a bug discovered by them.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2020-02-01_21:00:27-rados-wip-sage2-testing-2020-02-01-1055-distro-basic-smithi/ passing tests!

**Comment by liewegas** [comment]
@sebastian-philipp ok to merge?

**Pull request merged**